modules/templates/templates/template-detail.tsx

```diff
@@ -0,0 +1,327 @@
+import { GlobalCore } from "@/core/module/module.types";
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import {
+  GenesisTemplate,
+  GenesisTemplateField,
+} from "@/core/module/core.types";
+import { useTranslation } from "react-i18next";
+import styles from "./styles/template-detail.module.css";
+import { FaEdit, FaSave, FaTrash, FaArrowLeft, FaPlus } from "react-icons/fa";
+import MessageHandler from "@/core/message-handler";
+import DeleteConfirmation from "@/resources/containers/delete-confirmation";
+import Table from "@/resources/table/table";
+import FieldModal from "@/resources/containers/field-modal";
+import { useService } from "@/core/module/service.factory";
+
+const messageHandler = MessageHandler.get();
+type TableDataType = GenesisTemplateField & { key: string; name: string };
+
+const TemplateDetail = () => {
+  const router = useRouter();
+  const { t } = useTranslation();
+  const templateService = useService("templates");
+  const { id } = router.query;
+  const [template, setTemplate] = useState<GenesisTemplate | null>(null);
+  const [editingField, setEditingField] = useState<string | null>(null);
+  const [editedValues, setEditedValues] = useState<{
+    [key: string]: GenesisTemplateField & { name: string };
+  }>({});
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [fieldToDelete, setFieldToDelete] = useState<string | null>(null);
+  const [isFieldModalOpen, setIsFieldModalOpen] = useState(false);
+  const [currentFieldKey, setCurrentFieldKey] = useState<string | null>(null);
+
+  const typeOptions = ["text", "number", "multiselect"];
```

Comment:
we have `select` and `multiselect` are different but the config is the same. `text` doesn't have any config yet.

This can be move as a enum in the constants file or core types